igvc_platform/src/imu/YostLabDriver.cpp

```diff
@@ -17,21 +19,60 @@ YostLabDriver::YostLabDriver(ros::NodeHandle& nh_, ros::NodeHandle& priv_nh_)
 //! Destructor
 YostLabDriver::~YostLabDriver()
 {
+  this->updater.broadcast(diagnostic_msgs::DiagnosticStatus::ERROR, "IMU Node Terminated");
 }
+
 void YostLabDriver::restoreFactorySettings()
 {
   this->SerialWriteString(RESTORE_FACTORY_SETTINGS);
 }
+
 const std::string YostLabDriver::getSoftwareVersion()
 {
+  this->flush();
   this->SerialWriteString(GET_FIRMWARE_VERSION_STRING);
   const std::string buf = this->SerialReadLine();
   ROS_INFO_STREAM(this->logger << "Software version: " << buf);
   return buf;
 }
 
+void YostLabDriver::imu_diagnostic(diagnostic_updater::DiagnosticStatusWrapper &stat)
+{
+  if (sensor_temp_ > 185)
+  {
+    stat.summary(diagnostic_msgs::DiagnosticStatus::WARN, "IMU temp too high");
+  }
+  else if (sensor_temp_ < -40)
+  {
+    stat.summary(diagnostic_msgs::DiagnosticStatus::WARN, "IMU temp too low");
+  }
+  else if ((ros::Time::now() - lastUpdateTime_).toSec() > 1)
+  {
+    stat.summary(diagnostic_msgs::DiagnosticStatus::ERROR, "IMU not updating");
+  }
+  else if (quaternion_length_ < 0.98 || quaternion_length_ > 1.02)
```

Comment:
Also make these magic numbers not magic numbers, ie. `abs(quaternion_length - 1) > X` or something